Error unable to connect to socket 'queue/db/wdb' in Centos 7 manager for v4.9.0 Alpha 2

## Description

During the analysis performed in https://github.com/wazuh/wazuh/issues/24576, we detected an error on the OracleLinux 6 tests execution, the error itself is on the wazuh-manager (centos 7 system)

**Error displayed:**

```
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'
```

Checking the logs we can validate that `analysisd` starts after `wazuh-db`. The reason of the error is not clear, further research must be done. 

**Relevant logs**
```

2024/07/15 08:39:49 wazuh-db: INFO: Started (pid: 1373).
2024/07/15 08:39:51 wazuh-analysisd: INFO: Started (pid: 1398).
2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'.
```

**Affected pipeline:** https://ci.wazuh.info/job/Test_install/164927/consoleFull

@juliancnn ,

> Do we have the VMs up? 
- No, unfortunately they are docker containers running on AWS ECS

> ossec.log
- The pipeline does not provide this file as artifact, just a small part of it as stdout while doing some asserts

> tree of dir and permissions ls -lha /var/ossec/wazuh/queue/**

- This info is not currently available due to pipeline limitations


> Current status of service journalctl status wazuh-manager

- This info is not currently available due to pipeline limitations


> wazuh process list ps -aux | grep -i wazuh

- This info is not currently available due to pipeline limitations

> If you do not have this data, it would be useful to have the vagrant files used, to be able to replicate the environment and see if it is something recurring.

Docker images could be tricky to run locally

IMHO I suggest using the following information to make a first validation of a possible race condition or startup crash. More information will be shared by DM

```
05:42:56  TASK [test-manager : Installation test manager (Pytest)] ***********************
05:42:56  fatal: [amd64-centos-7-base_0]: FAILED! => {
05:42:56      "changed": true,
05:42:56      "cmd": "/usr/testing/bin/pytest --tb=short install_manager_test.py --to_version 4.9.0 --os 'centos 7' -v",
05:42:56      "delta": "0:00:00.919338",
05:42:56      "end": "2024-07-15 08:42:56.368661",
05:42:56      "rc": 1,
05:42:56      "start": "2024-07-15 08:42:55.449323"
05:42:56  }
05:42:56  
05:42:56  STDOUT:
05:42:56  
05:42:56  ============================= test session starts ==============================
05:42:56  platform linux -- Python 3.8.2, pytest-5.4.1, py-1.11.0, pluggy-0.13.1 -- /usr/testing/bin/python3.8
05:42:56  cachedir: .pytest_cache
05:42:56  rootdir: /install_manager
05:42:56  collecting ... collected 14 items
05:42:56  
05:42:56  install_manager_test.py::test_check_version PASSED                       [  7%]
05:42:56  install_manager_test.py::test_check_process_clusterd PASSED              [ 14%]
05:42:56  install_manager_test.py::test_check_process_modulesd PASSED              [ 21%]
05:42:56  install_manager_test.py::test_check_process_monitord PASSED              [ 28%]
05:42:56  install_manager_test.py::test_check_process_logcollector PASSED          [ 35%]
05:42:56  install_manager_test.py::test_check_process_remoted PASSED               [ 42%]
05:42:56  install_manager_test.py::test_check_process_syscheckd PASSED             [ 50%]
05:42:56  install_manager_test.py::test_check_process_analysisd PASSED             [ 57%]
05:42:56  install_manager_test.py::test_check_process_maild PASSED                 [ 64%]
05:42:56  install_manager_test.py::test_check_process_execd PASSED                 [ 71%]
05:42:56  install_manager_test.py::test_check_process_wazuhdb PASSED               [ 78%]
05:42:56  install_manager_test.py::test_check_log_errors FAILED                    [ 85%]
05:42:56  install_manager_test.py::test_check_ossec_conf PASSED                    [ 92%]
05:42:56  install_manager_test.py::test_connection_manager PASSED                  [100%]
05:42:56  
05:42:56  =================================== FAILURES ===================================
05:42:56  ____________________________ test_check_log_errors _____________________________
05:42:56  install_manager_test.py:128: in test_check_log_errors
05:42:56      assert expected_error == True, assert_description['error']['exception_not_detected']
05:42:56  E   AssertionError: An error was found in the log
05:42:56  E   assert False == True
05:42:56  E     +False
05:42:56  E     -True
05:42:56  ----------------------------- Captured stdout call -----------------------------
05:42:56  2024/07/15 08:39:44 wazuh-modulesd:router: INFO: Loaded router module.
05:42:56  2024/07/15 08:39:44 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
05:42:56  2024/07/15 08:39:48 wazuh-csyslogd: INFO: Remote syslog server not configured. Clean exit.
05:42:56  2024/07/15 08:39:48 wazuh-dbd: INFO: Database not configured. Clean exit.
05:42:56  2024/07/15 08:39:48 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
05:42:56  2024/07/15 08:39:48 wazuh-agentlessd: INFO: Not configured. Exiting.
05:42:56  2024/07/15 08:39:48 wazuh-authd: INFO: Started (pid: 1356).
05:42:56  2024/07/15 08:39:48 wazuh-authd: INFO: Accepting connections on port 1515. No password required.
05:42:56  2024/07/15 08:39:48 wazuh-authd: INFO: Setting network timeout to 1.000000 sec.
05:42:56  2024/07/15 08:39:49 wazuh-db: INFO: Started (pid: 1373).
05:42:56  2024/07/15 08:39:50 wazuh-execd: INFO: Started (pid: 1386).
05:42:56  2024/07/15 08:39:51 wazuh-analysisd: INFO: Total rules enabled: '7006'
05:42:56  2024/07/15 08:39:51 wazuh-analysisd: INFO: Started (pid: 1398).
05:42:56  2024/07/15 08:39:51 wazuh-analysisd: INFO: Cannot find 'queue/db/wdb'. Waiting 1 seconds to reconnect.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: Started (pid: 1412).
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/boot', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6000): Starting daemon...
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
05:42:56  2024/07/15 08:39:51 rootcheck: INFO: Starting rootcheck scan.
05:42:56  2024/07/15 08:39:51 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
05:42:56  2024/07/15 08:39:52 wazuh-analysisd: INFO: Cannot find 'queue/db/wdb'. Waiting 2 seconds to reconnect.
05:42:56  2024/07/15 08:39:52 wazuh-remoted: INFO: Started (pid: 1431). Listening on port 1514/TCP (secure).
05:42:56  2024/07/15 08:39:53 wazuh-remoted: INFO: Cannot find 'queue/db/wdb'. Waiting 1 seconds to reconnect.
05:42:56  2024/07/15 08:39:54 wazuh-remoted: INFO: Cannot find 'queue/db/wdb'. Waiting 2 seconds to reconnect.
05:42:56  2024/07/15 08:39:54 wazuh-logcollector: INFO: Monitoring output of command(360): df -P
05:42:56  2024/07/15 08:39:54 wazuh-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
05:42:56  2024/07/15 08:39:54 wazuh-logcollector: INFO: Monitoring full output of command(360): last -n 20
05:42:56  2024/07/15 08:39:54 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
05:42:56  2024/07/15 08:39:54 wazuh-logcollector: INFO: Started (pid: 1443).
05:42:56  2024/07/15 08:39:54 wazuh-analysisd: INFO: Cannot find 'queue/db/wdb'. Waiting 3 seconds to reconnect.
05:42:56  2024/07/15 08:39:55 wazuh-db: INFO: Created Global database backup "backup/db/global.db-backup-2024-07-15-08:39:55.gz"
05:42:56  2024/07/15 08:39:55 wazuh-monitord: INFO: Started (pid: 1475).
05:42:56  2024/07/15 08:39:56 wazuh-remoted: INFO: (1410): Reading authentication keys file.
05:42:56  2024/07/15 08:39:56 wazuh-logcollector: INFO: (9203): Monitoring journal entries.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:router: INFO: Loaded router module.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd: INFO: Started (pid: 1519).
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:agent-upgrade: INFO: (8153): Module Agent Upgrade started.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:router: INFO: Starting router module.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:task-manager: INFO: (8200): Module Task Manager started.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:content_manager: INFO: Starting content_manager module.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:database: INFO: Module started.
05:42:56  2024/07/15 08:39:56 sca: INFO: Module started.
05:42:56  2024/07/15 08:39:56 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_centos7_linux.yml'
05:42:56  2024/07/15 08:39:56 sca: INFO: Starting Security Configuration Assessment scan.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:download: INFO: Module started.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:control: INFO: Starting control thread.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:vulnerability-scanner: INFO: Starting vulnerability_scanner module.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:syscollector: INFO: Module started.
05:42:56  2024/07/15 08:39:56 wazuh-modulesd:syscollector: INFO: Starting evaluation.
05:42:56  2024/07/15 08:39:56 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_centos7_linux.yml'
05:42:56  2024/07/15 08:39:57 wazuh-modulesd:syscollector: INFO: Evaluation finished.
05:42:56  2024/07/15 08:39:57 indexer-connector: WARNING: No username and password found in the keystore, using default values.
05:42:56  2024/07/15 08:39:57 indexer-connector: WARNING: IndexerConnector initialization failed for index 'wazuh-states-vulnerabilities-ip-REDACTED', retrying until the connection is successful.
05:42:56  2024/07/15 08:39:57 wazuh-modulesd:vulnerability-scanner: INFO: Starting database file decompression.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Unable to connect to socket 'queue/db/wdb'
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Response from the Mitre database cannot be parsed.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: ERROR: Mitre matrix information could not be loaded.
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: INFO: (7200): Logtest started
05:42:56  2024/07/15 08:39:57 wazuh-analysisd: INFO: EPS limit disabled
05:42:56  2024/07/15 08:39:58 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
05:42:56  2024/07/15 08:39:58 wazuh-syscheckd: INFO: FIM sync module started.
05:42:56  2024/07/15 08:40:27 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_centos7_linux.yml'
05:42:56  2024/07/15 08:40:27 sca: INFO: Security Configuration Assessment scan finished. Duration: 31 seconds.
05:42:56  2024/07/15 08:40:39 rootcheck: INFO: Ending rootcheck scan.
05:42:56  2024/07/15 08:41:02 wazuh-authd: INFO: New connection from REDACTED
05:42:56  2024/07/15 08:41:02 wazuh-authd: INFO: Received request for a new agent (REDACTED) from: REDACTED
05:42:56  2024/07/15 08:41:02 wazuh-authd: INFO: Agent key generated for 'REDACTED' (requested by any)
05:42:56  2024/07/15 08:41:06 wazuh-remoted: INFO: (1409): Authentication file changed. Updating.
05:42:56  2024/07/15 08:41:06 wazuh-remoted: INFO: (1410): Reading authentication keys file.
05:42:56  2024/07/15 08:41:25 wazuh-analysisd: WARNING: Mitre Technique ID 'T1562.001' not found in database.
05:42:56  
05:42:56  =========================== short test summary info ============================
05:42:56  FAILED install_manager_test.py::test_check_log_errors - AssertionError: An er...
05:42:56  ========================= 1 failed, 13 passed in 0.20s =========================
05:42:56  
05:42:56  
05:42:56  MSG:
05:42:56  
05:42:56  non-zero return code
```

--- a/src/analysisd/mitre.c
+++ b/src/analysisd/mitre.c
@@ -61,6 +61,14 @@ int mitre_load() {
     os_calloc(OS_SIZE_6144 + 1, sizeof(char), wazuhdb_query);
     os_calloc(OS_MAXSTR, sizeof(char), response);
 
+    /* Connect to wdb */
+    sock = wdbc_connect_with_attempts(5);
+    if (sock < 0) {
+        merror("Unable to connect to Wazuh-DB for Mitre matrix information.");
+        result = -1;
+        goto end;
+    }
+
     /* Getting technique ID and name from Mitre's database in Wazuh-DB  */
     snprintf(wazuhdb_query, OS_SIZE_6144, SQL_GET_ALL_TECHNIQUES, MAX_TECHNIQUES_REQUEST, offset);
     techniques_json = wdbc_query_parse_json(&sock, wazuhdb_query, response, OS_MAXSTR);

--- a/src/headers/wazuhdb_op.h
+++ b/src/headers/wazuhdb_op.h
@@ -31,6 +31,7 @@ typedef enum wdbc_result {
 extern const char* WDBC_RESULT[];
 
 int wdbc_connect();
+int wdbc_connect_with_attempts(int max_attempts);
 int wdbc_query(const int sock, const char *query, char *response, const int len);
 int wdbc_query_ex(int *sock, const char *query, char *response, const int len);
 int wdbc_parse_result(char *result, char **payload);

--- a/src/shared/wazuhdb_op.c
+++ b/src/shared/wazuhdb_op.c
@@ -18,15 +18,25 @@
  * @return Socket descriptor or -1 if error.
  */
 int wdbc_connect() {
+    return wdbc_connect_with_attempts(3);
+}
 
-    int attempts;
+/**
+ * @brief Connects to Wazuh-DB socket with a maximum number of attempts.
+ *
+ * @param max_attempts Maximum number of attempts to connect.
+ * @return int Socket descriptor or -1 if error.
+ */
+int wdbc_connect_with_attempts(int max_attempts) {
+
+    assert(max_attempts > 0);
     int wdb_socket = -1;
     char sockname[PATH_MAX + 1];
 
     strcpy(sockname, WDB_LOCAL_SOCK);
 
 
-    for (attempts = 1; attempts <= 3 && (wdb_socket = OS_ConnectUnixDomain(sockname, SOCK_STREAM, OS_SIZE_6144)) < 0; attempts++) {
+    for (int attempts = 1; attempts <= max_attempts && (wdb_socket = OS_ConnectUnixDomain(sockname, SOCK_STREAM, OS_SIZE_6144)) < 0; attempts++) {
         switch (errno) {
         case ENOENT:
             minfo("Cannot find '%s'. Waiting %d seconds to reconnect.", sockname, attempts);

--- a/src/unit_tests/analysisd/CMakeLists.txt
+++ b/src/unit_tests/analysisd/CMakeLists.txt
@@ -64,7 +64,8 @@ list(APPEND analysisd_names "test_labels")
 list(APPEND analysisd_flags "-Wl,--wrap,wdb_get_agent_labels")
 
 list(APPEND analysisd_names "test_mitre")
-list(APPEND analysisd_flags "-Wl,--wrap,wdbc_query_ex -Wl,--wrap,wdbc_query_parse_json ${HASH_OP_WRAPPERS} ${DEBUG_OP_WRAPPERS}")
+list(APPEND analysisd_flags "-Wl,--wrap,wdbc_connect_with_attempts -Wl,--wrap,wdbc_query_ex \
+                             -Wl,--wrap,wdbc_query_parse_json ${HASH_OP_WRAPPERS} ${DEBUG_OP_WRAPPERS}")
 
 list(APPEND analysisd_names "test_rules")
 LIST(APPEND analysisd_flags "-Wl,--wrap,_mwarn -Wl,--wrap,_merror -Wl,--wrap,_os_analysisd_add_logmsg \

--- a/src/unit_tests/analysisd/test_mitre.c
+++ b/src/unit_tests/analysisd/test_mitre.c
@@ -57,11 +57,10 @@ void test_queryid_error_socket(void **state)
     int ret;
     cJSON * id_array = NULL;
 
-    will_return(__wrap_wdbc_query_parse_json, -2);
-    will_return(__wrap_wdbc_query_parse_json, id_array);
+    will_return(__wrap_wdbc_connect_with_attempts, -2);
 
-    expect_string(__wrap__merror, formatted_msg, "Unable to connect to socket 'queue/db/wdb'");
-    expect_string(__wrap__merror, formatted_msg, "Response from the Mitre database cannot be parsed.");
+
+    expect_string(__wrap__merror, formatted_msg, "Unable to connect to Wazuh-DB for Mitre matrix information.");
     expect_string(__wrap__merror, formatted_msg, "Mitre matrix information could not be loaded.");
 
     ret = mitre_load();
@@ -75,6 +74,7 @@ void test_queryid_no_response(void **state)
     int ret;
     cJSON * id_array = NULL;
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, -1);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -94,6 +94,8 @@ void test_queryid_bad_response(void **state)
     cJSON * id_array = NULL;
 
     char *response_ids = "err not found";
+
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 1);
     will_return(__wrap_wdbc_query_parse_json, response_ids);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -113,6 +115,7 @@ void test_queryid_error_parse(void **state)
     int ret;
     cJSON * id_array = cJSON_Parse("[{\"id\":}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -130,6 +133,7 @@ void test_queryid_empty_array(void **state)
     int ret;
     cJSON * id_array = cJSON_Parse("[ ]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -145,6 +149,7 @@ void test_queryid_error_parse_technique_id(void **state) {
     int ret;
     cJSON * id_array = cJSON_Parse("[{\"ids\":\"technique-0001\"},{\"ids\":\"technique-0002\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -166,6 +171,7 @@ void test_queryid_error_parse_technique_name(void **state) {
     int ret;
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\"},{\"id\":\"technique-0002\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -187,6 +193,7 @@ void test_queryid_error_parse_technique_external_id(void **state) {
     int ret;
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
 
@@ -210,6 +217,7 @@ void test_querytactics_error_socket(void **state) {
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\",\"external_id\":\"T1001\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\",\"external_id\":\"T1002\"}]");
     cJSON * tactic_array = NULL;
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -238,6 +246,7 @@ void test_querytactics_no_response(void **state) {
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\",\"external_id\":\"T1001\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\",\"external_id\":\"T1002\"}]");
     cJSON * tactic_array = NULL;
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -267,6 +276,7 @@ void test_querytactics_bad_response(void **state) {
     cJSON * tactic_array = NULL;
     char * response_tactics = "err not found";
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -296,6 +306,7 @@ void test_querytactics_error_parse(void **state) {
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\",\"external_id\":\"T1001\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\",\"external_id\":\"T1002\"}]");
     cJSON * tactic_array = cJSON_Parse("[{\"phase_name\":}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -323,6 +334,7 @@ void test_querytactics_empty_array(void **state) {
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\",\"external_id\":\"T1001\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\",\"external_id\":\"T1002\"}]");
     cJSON * tactic_array = cJSON_Parse("[ ]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -350,6 +362,7 @@ void test_querytactics_error_parse_tactics(void **state) {
     cJSON * id_array = cJSON_Parse("[{\"id\":\"technique-0001\",\"name\":\"Technique1\",\"external_id\":\"T1001\"},{\"id\":\"technique-0002\",\"name\":\"Technique2\",\"external_id\":\"T1002\"}]");
     cJSON * tactic_array = cJSON_Parse("[{\"phase\":\"Discovery\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -378,6 +391,7 @@ void test_queryname_error_socket(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = NULL;
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -411,6 +425,7 @@ void test_queryname_no_response(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = NULL;
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -445,6 +460,7 @@ void test_queryname_bad_response(void **state) {
     cJSON * tactic_info_array = NULL;
     char * response_tactics = "err not found";
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -479,6 +495,7 @@ void test_queryname_error_parse(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = cJSON_Parse("[{\"info\":}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -511,6 +528,7 @@ void test_queryname_error_parse_technique_name(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = cJSON_Parse("[{\"info\":\"Tactic1\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -543,6 +561,7 @@ void test_queryname_error_parse_technique_external_id(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = cJSON_Parse("[{\"name\":\"Tactic1\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -575,6 +594,7 @@ void test_query_tactics_error_filling_technique(void **state) {
     cJSON * tactic_array = cJSON_Parse("[{\"tactic_id\":\"tactic-0001\"}]");
     cJSON * tactic_info_array = cJSON_Parse("[{\"name\":\"Tactic1\",\"external_id\":\"TA001\"}]");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);
@@ -612,6 +632,7 @@ void test_query_tactics_success(void **state) {
     cJSON * tactic_info_array = cJSON_Parse("[{\"name\":\"Tactic1\",\"external_id\":\"TA001\"}]");
     cJSON * technique_last = cJSON_Parse(" ");
 
+    will_return(__wrap_wdbc_connect_with_attempts, 1);
     /* Mitre's techniques IDs query */
     will_return(__wrap_wdbc_query_parse_json, 0);
     will_return(__wrap_wdbc_query_parse_json, id_array);

--- a/src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.c
+++ b/src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.c
@@ -179,6 +179,14 @@ void __wrap_wdbi_update_completion(__attribute__((unused))wdb_t * wdb,
     check_expected(manager_checksum);
 }
 
+int __wrap_wdbc_connect_with_attempts(__attribute__((unused)) int max_attempts) {
+
+    if (max_attempts <= 0) {
+        fail_msg("Attempts must be greater than 0.");
+    }
+    return mock();
+}
+
 cJSON* __wrap_wdbc_query_parse_json(__attribute__((unused)) int *sock,
                                     __attribute__((unused)) const char *query,
                                     char *response,

--- a/src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.h
+++ b/src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.h
@@ -57,6 +57,8 @@ int __wrap_wdbi_query_checksum(wdb_t *wdb, wdb_component_t component, const char
 
 int __wrap_wdbi_query_clear(wdb_t *wdb, wdb_component_t component, const char *payload);
 
+int __wrap_wdbc_connect_with_attempts(int max_attempts);
+
 cJSON* __wrap_wdbc_query_parse_json(int *sock, const char *query, char *response, const int len);
 
 wdbc_result __wrap_wdbc_query_parse(int *sock, const char *query, char *response, const int len, char** payload);

